### ISSUE TITLE
Add ignorePattern: "" option to max-line-length

### ISSUE THREAD

**Issue by handonam** [stylelint/stylelint#2219]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Using max-line-length with a long import statement causes the error to be thrown.  Should have an option to ignore patterns (much like eslint, with `{ "ignorePattern": "^import\\s.+\\sfrom\\s.+;$" }],`), or some way to ignore `@imports`.

> Which rule, if any, is this issue related to?

`max-line-length`

> What CSS is needed to reproduce this issue?

```css
@import (reference) "some/really/long/path/to/a/file/that/is/over/the/rainbow/or/something/likethat.scss";
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "max-line-length": 100,
  }
}
```

> Which version of stylelint are you using?

`7.7.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

atom plugin / PostCSS Plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No.

> What did you expect to happen?

Ignore imports when something like ignorePattern is specified

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
Expected line length to be no more than 100 characters
```

**Comment by ntwb** [comment]
Thanks for creating this issue and using our template @handonam 👍 

I've confirmed the behaviour you describe though it is only when using `@import "something"`, if including the `url` in the statement such as `@import url(something)` then this rule works as expected. I've added some tests in https://github.com/stylelint/stylelint/pull/2221/commits/480c3fc01db30a748cd900eb3ab640c5a18e661d confirming this.

Currently `max-line-length` [uses the regex](https://github.com/stylelint/stylelint/blob/master/lib/rules/max-line-length/index.js#L37) `/url\(.*\)` to detect URLs, thus it doesn't detect the valid use of `@import` without the URL string.

Via https://developer.mozilla.org/en-US/docs/Web/CSS/@import :
```css
@import url("fineprint.css") print;
@import url("bluish.css") projection, tv;
@import 'custom.css';
@import url("chrome://communicator/skin/");
@import "common.css" screen, projection;
@import url('landscape.css') screen and (orientation:landscape)
```

Rather than adding an ignore option to the rule I've updated the rule regex adding support to also detect `@import "something"` in https://github.com/stylelint/stylelint/pull/2221/commits/926e35e5e4628235aef938ef92c8517f2ddc9094

**Comment by evilebottnawi** [comment]
@ntwb we can add `ignorePattern` as doing `eslint` (http://eslint.org/docs/rules/max-len), this makes it possible to cover most related issue
